ui/src/timeMachine/actions/queries.ts

```diff
@@ -212,9 +221,11 @@ const saveDraftQueries = (): SaveDraftQueriesAction => ({
   type: 'SAVE_DRAFT_QUERIES',
 })
 
-export const saveAndExecuteQueries = () => dispatch => {
+export const saveAndExecuteQueries = (
+  abortController?: AbortController
+) => dispatch => {
   dispatch(saveDraftQueries())
-  dispatch(executeQueries())
```

Comment:
could you dispatch(setQueryToLoading) here instead of as part of the `  private handleClick = (): void => {` method in submit button?

Reply:
That sounds great!